<launch>
    <arg name="keypoint_classifier_label" default="$(find hand_gesture_recognition)/src/model/keypoint_classifier/keypoint_classifier_label.csv"/>
    <arg name="keypoint_classifier_model" default="$(find hand_gesture_recognition)/src/model/keypoint_classifier/keypoint_classifier.tflite"/>
    
    <arg name="pose_2d_detect"            default="true"/>
    <arg name="pose_2d_log_show"          default="true"/>
    <arg name="pose_2d_img_show"          default="true"/>
    <arg name="pose_2d_img_pub"           default="true"/>

    <arg name="sub_img_topic_name"        default="/camera/rgb/image_raw"/>

    <node name="hand_sign_recognition" pkg="hand_gesture_recognition" type="hand_sign_recognition.py" output="screen">
        <param name="keypoint_classifier_label" type="string" value="$(arg keypoint_classifier_label)" />
        <param name="keypoint_classifier_model" type="string" value="$(arg keypoint_classifier_model)" />

        <param name="pose_2d_detect"     type="bool" value="$(arg pose_2d_detect)" />
        <param name="pose_2d_log_show"   type="bool" value="$(arg pose_2d_log_show)" />
        <param name="pose_2d_img_show"   type="bool" value="$(arg pose_2d_img_show)" />
        <param name="pose_2d_img_pub"    type="bool" value="$(arg pose_2d_img_pub)" />

        <param name="sub_img_topic_name" type="string" value="$(arg sub_img_topic_name)" />
    </node>
</launch>
